src/librustc_error_codes/lib.rs

```diff
@@ -0,0 +1,26 @@
+//! This library is used to gather all error codes into one place. The goal
+//! being to make their maintenance easier.
+
+#[macro_export]
+macro_rules! register_diagnostics {
+    ($($ecode:ident: $message:expr,)*) => (
```

Comment:
This matcher looks unused now.

Reply:
It's still being used actually. (it's the separation between long and short error codes) :)